### DISCUSSION TITLE
Add enum to profiles table from raw_user_meta_data

### DISCUSSION THREAD

**Discussion by andyjamesn** [supabase/supabase#9537] [Questions] [answered]
I have an enum for gender

`create type public.gender as enum ('male','female')`

I am creating a user and on the front end they set a gender which is sent along as metadata with signup 

```
data: {
    username: username.value,
    age: 23,
    gender: gender.value,
}
```

This is stored in the users metadata `{"age":23,"gender":"female","username":"test user"}`

I have a function and trigger that populates the profiles table

```
begin
  insert into public.profiles (id, username, gender)
  values (new.id, new.raw_user_meta_data->>'username', new.raw_user_meta_data->>'gender');
  return new;
end;
```

Using the above function throws an error. 

`column "gender" is of type public.gender but expression is of type text`

Clearly this is because in the metadata gender is a string. How do I edit my function to set the correct type for gender.

**Comment by GaryAustin1** [comment, marked as answer]
So if your enum type is named gender then you would do `(new.raw_user_meta_data->>'gender')::gender`   The 2nd gender is the name of the enum type.

**Reply by andyjamesn** [reply]
Thanks this worked but I had to tweak it slightly as my enum is specifically set to public.gender

`(new.raw_user_meta_data->>'gender')::public.gender`
